File: specter/ios/Sources/base/Pair.swift
import Foundation

public struct Pair<L, R> {
    public var left: L
    public  var right: R
    
    public init(left: L, right: R) {
        self.left = left
        self.right = right
    }
}

extension Pair: Equatable where L: Equatable, R: Equatable {}
extension Pair: Hashable where L: Hashable, R: Hashable {}
extension Pair: Sendable where L: Sendable, R: Sendable {}

public typealias IdentifiableValue<ID, Value> = Pair<ID, Value>

public func makeIdentifiable<V>(value: V, id: UUID = UUID()) -> IdentifiableValue<UUID, V> {
    Pair(left: id, right: value)
}

File: polytop/rust/poly-tui/src/pair.rs

#[derive(Clone, Debug, Eq, PartialEq)]
pub struct Pair<L, R> {
    pub left: L,
    pub right: R,
}

impl<L, R> Pair<L, R> {
    pub fn new(left: L, right: R) -> Self {
        Self { left, right }
    }
}